app/ui/src/main/java/de/mm20/launcher2/ui/launcher/search/SearchVM.kt

```diff
@@ -165,34 +170,47 @@ class SearchVM : ViewModel(), KoinComponent {
                             }
                         }
 
-                        if (query.isNotEmpty() && launchOnEnter.value)  {
-                            bestMatch.value = listOf(
-                                apps,
-                                workApps,
-                                shortcuts,
-                                unitConv,
-                                calc,
-                                events,
-                                contacts,
-                                wikipedia,
-                                website,
-                                files,
-                                actions
-                            ).firstNotNullOfOrNull { it.firstOrNull() }
-                        }
+                        ranksByLaunchCount.collectLatest {
+
+                            for ((domain, ranks) in it.groupBy { it.domain }) {
```

Comment:
Do you think it would be a performance improvement if we already ordered the results at the database-level? E.g. `GROUP BY 'type'`? Or is kotlin already fast enough?